server_ftp/lib/abstract_ftp_server.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from abc import ABC, abstractmethod
+
+
+class AbstractFtpServer(ABC):
+    def __init__(self, url, user, password, port, path):
```

Comment:
Do not connect in the init method. just put self.server = None.

So connection is always done by calling the connect method.